### DISCUSSION TITLE
ModuleFederation 2.0 - Switch to RSPack

### DISCUSSION THREAD

**Discussion by Lonli-Lokli** [nrwl/nx#23261] [Feature Requests]
Could you please update Nx support for module federation with the new way (eg without Webpack5)?

https://module-federation.io/guide/start/index.html#-what-is-module-federation-20

**Comment by jaysoo** [comment]
We'll always need to support Webpack, but we have discussed MF 2.0 and adding rspack support. No timeline yet.

**Comment by soumyanildas** [comment]
I have done a setup with Rspack and Module Federation 2.0 but I have remotes as separate repos as that was our requirement.

Maybe you could try the same in a monorepo setup?

Check my blog here - https://blog.soumyanildas.com/nx-rspack-micro-frontend

But there's an issue with sass module only working on 18.0.6 `@nx/rspack`
